lib/components/molecules/Typeahead/Typeahead.js

```diff
@@ -0,0 +1,180 @@
+// @flow
+/**
+ *
+ * Typeahead
+ *
+ */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import classNames from 'classnames';
+
+import styles from './Typeahead.style';
+import type { Props } from './types';
+import Input from '../../atoms/Input';
+import List from '../List';
+import Label from '../../atoms/Label';
+import ListItem from './ListItem';
+import { ENTER_KEY, UP_ARROW, DOWN_ARROW, ESCAPE, TAB } from './constants';
+
+const Typeahead = ({
+  id,
+  getResults,
+  results,
+  className,
+  renderSuggestionItem,
+  searchValue,
+  onChangeHandler,
+  getSelectedSuggestionValue,
+  label,
+  isFieldLabel,
+  ...otherProps
+}: Props): React.Element<any> => {
+  const [updatedResults, setFilteredResults] = useState(results);
+  const [activeSuggestion, setActiveSuggestion] = useState(0);
+  const [showResults, setShowResults] = useState(true);
+
+  const triggerGetResults = val => {
+    return val && val.trim().length >= 3;
+  };
+
+  /**
+   * Function to handle item selection
+   * @param {'*'} e
+   */
+  const handleClick = item => {
+    setActiveSuggestion(0);
+    getSelectedSuggestionValue(item);
+    setShowResults(false);
+  };
+
+  const getFilteredResults = (allResults, inputText) => {
+    const filteredResults = allResults.filter(
+      suggestion => suggestion.toLowerCase().indexOf(inputText.toLowerCase()) > -1
+    );
+    return filteredResults;
+  };
+
+  /**
+   * Function to handle uparrow, downarrow, enter and escape
+   * @param {*} e
+   */
+  const handleKeyDown = e => {
+    switch (e.keyCode) {
+      case ENTER_KEY: {
+        setActiveSuggestion(0);
+        getSelectedSuggestionValue(updatedResults[activeSuggestion]);
+        setShowResults(false);
+        break;
+      }
+      case UP_ARROW: {
+        if (activeSuggestion === 0) {
+          setActiveSuggestion(updatedResults.length - 1);
+          return;
+        }
+        setActiveSuggestion(activeSuggestion - 1);
+        break;
+      }
+      case DOWN_ARROW: {
+        if (activeSuggestion + 1 === updatedResults.length) {
+          setActiveSuggestion(0);
+          return;
+        }
+        setActiveSuggestion(activeSuggestion + 1);
+        break;
+      }
+      case TAB:
+      case ESCAPE: {
+        setActiveSuggestion(0);
+        setShowResults(false);
+        break;
+      }
+      default:
+        break;
+    }
+  };
+
+  /**
+   * Function to handle text change in input
+   * @param {*} e
+   */
+  const handleChange = e => {
+    if (onChangeHandler) {
+      onChangeHandler(e);
+    }
+    // Note: Need to handle Debounce
+    const { value: inputText } = e.currentTarget;
+    if (triggerGetResults(inputText)) {
+      if (getResults) {
+        getResults();
+        setFilteredResults(results);
```

Comment:
Try avoid the duplication, in such function calls no point calling same function from both branches